containers/Token.ts

```diff
@@ -21,6 +21,7 @@ function useToken() {
   const [decimals, setDecimals] = useState<number | null>(null);
   const [allowance, setAllowance] = useState<number | "Infinity" | null>(null);
   const [balance, setBalance] = useState<number | null>(null);
+  const [balanceBN, setBalanceBN] = useState<BigNumber | null>(null);
```

Comment:
I pass the balance as a BN so that client React components can compare the raw BN instead of the truncated balance